src/components/TradeWidget/FormMessage.tsx

```diff
@@ -39,6 +46,45 @@ const FormMessage = styled.div.attrs<{ className?: string }>(props => ({
     box-sizing: border-box;
     margin: 0.3rem 0 1rem;
   }
+  &.hidden {
+    visibility: hidden;
+  }
 `
 
+interface FormInputErrorProps {
+  errorMessage?: string
+  schemaError?: boolean
+  visibilityHidden?: boolean
+  wrapperMargin?: string
+}
+
+export const FormInputError: React.FC<FormInputErrorProps> = ({
+  errorMessage,
+  schemaError = true,
+  visibilityHidden = true,
+  wrapperMargin = '0.5rem 0',
+}) => {
+  const message = useCallback(() => {
```

Comment:
Since you are calling `message()` every time anyway, no need for it to be a function, do `useMemo` and construct message inside

Reply:
A+